src/compiler/core_public.rs

```diff
@@ -0,0 +1,17 @@
+pub struct SortedArray<TItem> {
+    pub _vec: Vec<TItem>,
```

Comment:
Similar to the comment below, it seems bad to make this public, because anyone could mutate it, breaking the contract that a `SortedArray` is sorted.

Reply:
Good point, updated to make `_vec` private and for the one existing place that was accessing `_vec` (where it's trying to extract (a copy of) the inner `Vec`), I implemented `From<&SortedArray<TItem>> for Vec<TItem>` so that it can do `.into()` instead of `._vec.clone()`